loki/transformations/single_column/vector.py

```diff
@@ -450,6 +480,196 @@ def transform_subroutine(self, routine, **kwargs):
                     # Mark outer driver loops
                     self.mark_driver_loop(routine, loop)
 
+# alias for backwards compability
+SCCRevectorTransformation = SCCVecRevectorTransformation
+
+class SCCSeqRevectorTransformation(BaseRevectorTransformation):
+    """
+    A transformation to wrap thread-parallel IR sections within a horizontal loop
+    in a way that the horizontal loop is hoisted/moved to the driver level while
+    the horizontal/loop index is passed as an argument.
+    This transformation relies on markers placed by :any:`SCCDevectorTransformation`.
+    Parameters
+    ----------
+    horizontal : :any:`Dimension`
+        :any:`Dimension` object describing the variable conventions used in code
+        to define the horizontal data dimension and iteration space.
+    """
+
+    process_ignored_items = True
+
+    def remove_vector_sections(self, section):
+        """
+        Remove all thread-parallel :any:`Section` objects within a given
+        code section
+        Parameters
+        ----------
+        routine : :any:`Subroutine`
+            Subroutine to apply this transformation to.
+        section : tuple of :any:`Node`
+            Code section in which to replace vector-parallel
+            :any:`Section` objects.
+        """
+        # Wrap all thread-parallel sections into horizontal thread loops
+        mapper = {
+            s: s.body
+            for s in FindNodes(ir.Section).visit(section)
+            if s.label == 'vector_section'
+        }
+        return Transformer(mapper).visit(section)
+
+    def mark_vector_reductions(self, routine, section):
+        """
+        Mark vector-reduction loops in marked vector-reduction
+        regions.
+        If a region explicitly marked with
+        ``!$loki vector-reduction(<reduction clause>)``/
+        ``!$loki end vector-reduction`` is encountered, we replace
+        existing ``!$loki loop vector`` loop pragmas and add the
+        reduction keyword and clause. These will be turned into
+        OpenACC equivalents by :any:`SCCAnnotate`.
+        """
+        with pragma_regions_attached(routine):
+            for region in FindNodes(ir.PragmaRegion).visit(section):
+                if is_loki_pragma(region.pragma, starts_with='vector-reduction'):
+                    if (reduction_clause := re.search(r'reduction\([\w:0-9 \t]+\)', region.pragma.content)):
+
+                        loops = FindNodes(ir.Loop).visit(region)
+                        if loops:
+                            pragma = ir.Pragma(keyword='loki', content=f'loop vector {reduction_clause[0]}')
+                            # Update loop and region in place to remove marker pragmas
+                            loops[0]._update(pragma=(pragma,))
+                            region._update(pragma=None, pragma_post=None)
```

Comment:
Thanks for expanding the tests a bit. These 4 lines are still not hit in the test base - just pointing this out in case that is easy to fix, not making this a requirement here.